kustomize/rook-cluster/rook-cluster.yaml

```diff
@@ -77,7 +77,7 @@ spec:
   # enable prometheus alerting for cluster
   monitoring:
     # requires Prometheus to be pre-installed
-    enabled: true
+    enabled: false
```

Comment:
Since we have enabled monitoring on other helm charts we should also default to true here if possible

Reply:
+1 i think it makes sense to enable monitoring by default.

Reply:
Without a prometheus cluster stood up already this flag prevents the ceph cluster from being in a healthy state.

Reply:
Error message when flag is enabled without a pre-existing prometheus cluster [1].  We need to resolve by updating documentation and creating a prometheus cluster before ceph install or disable monitoring in rook-ceph

[1]

ceph-cluster-controller: failed to reconcile CephCluster "rook-ceph/rook-ceph". failed to reconcile cluster "rook-ceph": failed to configure local ceph cluster: failed to create cluster: failed to start ceph mgr: failed to enable mgr services: failed to enable service monitor: service monitor could not be enabled: failed to retrieve servicemonitor. servicemonitors.monitoring.coreos.com "rook-ceph-mgr" is forbidden: User "system:serviceaccount:rook-ceph:rook-ceph-system" cannot get resource "servicemonitors" in API group "monitoring.coreos.com" in the namespace "rook-ceph"

Reply:
You'll just need to deploy Prometheus in the existing cluster, either in k8s or cephadm. Cephadm should enable it by default from what I recall

Reply:
Maybe we could add something to kustomize when installing prometheus and enable logging in rook-ceph, we create a chicken and egg issue if we need a pvc for prometheus and we don't have a rook-ceph cluster serving storage.

Reply:
We might actually run Prometheus on local storage in such situation and run long term storage on ceph is needed but yes that will be something to lookout for

Reply:
We could maybe deploy the CRD's separately but they are part of the prometheus operator deploy today. I'm okay with disabling this by default given we'll be focusing on external ceph monitoring going forward and this would let people create labs again